includes/wp-admin/class-health-check.php

```diff
@@ -448,4 +455,46 @@ public static function test_pretty_permalinks() {
 
 		return $result;
 	}
+
+	/**
+	 * Surge integration test.
+	 *
+	 * @return array The test result.
+	 */
+	public static function test_surge_integration() {
+		$result = array(
+			'label'       => \__( 'Compatibility with Surge', 'activitypub' ),
+			'status'      => 'good',
+			'badge'       => array(
+				'label' => \__( 'ActivityPub', 'activitypub' ),
+				'color' => 'green',
+			),
+			'description' => \sprintf(
+				'<p>%s</p>',
+				\__( 'Surge is well configured to work with ActivityPub.', 'activitypub' )
+			),
+			'actions'     => '',
+			'test'        => 'test_surge_integration',
+		);
+
+		if ( ! \defined( 'WP_CACHE_CONFIG' ) ) {
+			$result['status']         = 'critical';
+			$result['label']          = \__( 'Surge might not be properly configured.', 'activitypub' );
+			$result['badge']['color'] = 'red';
+			$result['description']    = \sprintf(
+				'<p>%s</p>',
+				\__( 'Surge isn’t currently set up to work with ActivityPub. While this isn’t a major problem, it’s a good idea to enable support. Without it, some technical files (like JSON) might accidentally show up in your website’s cache and be visible to visitors.', 'activitypub' )
+			);
+			$result['actions']        = \sprintf(
+				'<p>%s</p>',
+				sprintf(
+					// translators: %s: Plugin directory path.
+					\__( 'To enable the ActivityPub integration with Surge, add the following line to your <code>wp-config.php</code> <file></file>: <br /><code>define( \'WP_CACHE_CONFIG\', %sintegration/surge.php );</code>', 'activitypub' ),
```

Comment:
```suggestion
					\__( 'To enable the ActivityPub integration with Surge, add the following line to your <code>wp-config.php</code> file: <br /><code>define( \'WP_CACHE_CONFIG\', %sintegration/surge.php );</code>', 'activitypub' ),
```

Let's [attempt to do it for them](https://github.com/kovshenin/surge/blob/b9bbb79421a49761cf42f6dcb81cd1b5fad3c8db/include/install.php#L36-L63) before giving them a task?

Reply:
I tried. This is needed so early in the process, that it hast to be added to the `wp-config.php`. Every place in the ActivityPub plugin was too late.

Reply:
It doesn't have to be there on the first page load. We could attempt to write it to the file and keep track of the attempt with an option. If the option is set and the constant doesn't exist, we'd then show the check?

Reply:
Is this edge case worth the work?

Reply:
Which edge case?

I'm proposing to attempt to do write that constant definition to wp-config, so users don't have to do it manually. The Surge plugin does the same with the WP_CACHE constant, so I assume the chances are high we could do the same with WP_CACHE_CONFIG.

Reply:
Edge Case: Not a lot people might use both plugins

Reply:
But maybe a nice POC that could be reused by other plugins... I will give it a try...